pallets/xvm/src/lib.rs

```diff
@@ -231,7 +254,6 @@ impl<T: Config> Pallet<T> {
         let weight_limit = context
             .weight_limit
             .saturating_sub(WeightInfoOf::<T>::wasm_call_overheads());
-        let value = Default::default();
 
         // Note the skip execution check should be exactly before `pallet_contracts::bare_call`
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `weight_limit` calculation has been updated to subtract the overheads specific to `wasm_call`. LGTM!



<!-- This is an auto-generated comment by OSS CodeRabbit -->